run_inference.py

```diff
@@ -1,31 +1,92 @@
-from transformers import AutoModelForCausalLM, AutoTokenizer
-from prompt_template import get_prompt_template_by_name
 import torch
 import typer
+import requests
+from typing import Union
+from bs4 import BeautifulSoup
+from prompt_template import get_prompt_template_by_name
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+
+def crawl_desc_leetcode_url(url: str) -> Union[str, None]:
+    """Crawl algorithm description from leetcode"""
+    url = url.strip()
+    if "leetcode.com" not in url:
+        raise ValueError(
+            "Please input url from leetcode. "
+            "For example: 'https://leetcode.com/problems/regular-expression-matching/"
+        )
+    response = requests.get(url)
+    response.raise_for_status()
+    response.encoding = "utf-8"
+    html = response.text
+    if html:
+        soup = BeautifulSoup(html, "html.parser")
+
+        # Find the meta tag with name="description"
+        meta_description = soup.find("meta", attrs={"name": "description"})
+
+        # Extract the content attribute value
+        if meta_description:
+            content_value = meta_description.get("content")
+            return content_value
+        else:
+            return None
+    else:
+        return None
+
+
+def main(leetcode_url: str = None, input_file: str = None, model_path: str = "khaimaitien/leetcode_solver_7b"):
```

Comment:
Can you change this to:
leetcode_url: str = typer.Option("")
input_file: str = typer.Option("")